### ISSUE TITLE
AP => ATProto: Allow link previews for bsky.app URLs

### ISSUE THREAD

**Issue by chaperon1234** [snarfed/bridgy-fed#1597]
https://github.com/snarfed/bridgy-fed/issues/1591
This is a continuation of the following.

I have verified a few things and I don't think this issue has been resolved yet.
It seems that the link preview does and does not open for some URLs.

Link opens: (OK case)
https://bsky.app/profile/chaperon.fedibird.com.ap.brid.gy/post/3lcen2o2fmqb2

Link does not open: (NG case)
https://bsky.app/profile/chaperon.fedibird.com.ap.brid.gy/post/3lchk7mg7apq2
https://bsky.app/profile/chaperon.fedibird.com.ap.brid.gy/post/3lcj7maovwda2
https://bsky.app/profile/chaperon.fedibird.com.ap.brid.gy/post/3lcf44xx4ueh2

Of the failing cases, I thought that the last URL might be because the account owner has selected a setting not to be displayed from logged out users. But the other two should be displayed? I think.

**Comment by snarfed** [comment]
Ah, these are all bsky.app URLs. We don't currently generate link previews for bsky.app URLs; we expect that people will @-mention users and quote-post posts. We could reconsider that! But it's not a high priority right now. Sorry.

**Comment by chaperon1234** [comment]
Thanks for the reply.

> people will @-mention users and quote-post posts.

How do I describe this from the Fediverce side?
It would be helpful if it means that users who are not bridged can quote me, is that possible?

**Comment by snarfed** [comment]
> It would be helpful if it means that I can quote from my side to users who are not bridged, is that possible?

Hmmm, no, you can only quote post or @-mention users who exist in your network, either native or bridged.

**Comment by chaperon1234** [comment]
"Hey, these illustrations are great, look at them guys!”
I have a desire to repost the person's posted image in link previews, etc. in that sense.

I would be happy to see the image in the link preview, so I would appreciate it if you could consider this.

**Comment by Tamschi** [comment]
I think that showing previews for Bluesky post links may be a bad idea.
It's like a quote that circumvents Bluesky's quote safety tooling and prevents the linked-to user from removing their content.

**Comment by snarfed** [comment]
@Tamschi I don't disagree!...but as a counterpoint, we don't currently try to blocklist links to fediverse posts, we happily generate link previews for them. It would obviously take more effort to detect and block them, something like #552.

**Comment by snarfed** [comment]
> Could we turn a Bluesky post link that stands alone on the last line into a quote (while removing the text link)?

Shipped this a bit ago!

**Comment by chaperon1234** [comment]
Yes, that's right. I've noticed this feature helping me out for a while now. Thanks for your response!

**Comment by MarkWhybird** [comment]
Laffy (@gottalaff.bsky.social@bsky.brid.gy) is a Mastodon icon. His bluesky feed is 95% reposts of selected useful bluesky posts, often with just one or two words added. Unfortunately it comes through looking like this:
<img width="1206" height="2622" alt="Image" src="https://github.com/user-attachments/assets/a2cd0ed4-1830-43ce-9c02-ab7d30da60f2" /> (viewed here in Mona, but it’s the same in other clients). Is there something Laffy can easily do to make them show better? (p.s. discussion on Mastodon is here: https://aus.social/@whybird/114854182452950916)

**Comment by snarfed** [comment]
Hi @MarkWhybird! This issue is for the opposite direction, for fediverse posts that are bridged into Bluesky.

For Bluesky posts bridged into the fediverse, like you're seeing here, Bridgy Fed doesn't (can't) generate link previews. Each fediverse instance generates those link previews itself...and oddly, Mastodon doesn't seem to generate previews for bsky.app links, even though those pages do have the right metadata. That seems to be a Mastodon-wide thing, not limited to any individual instance. I'm not sure why. Maybe bsky.app is on a blocklist built into the software, or maybe something else?

**Comment by MarkWhybird** [comment]
Ah, ok. Thanks; much appreciated.

**Comment by renchap** [comment]
There is nothing in the Mastodon software that prevents displaying link previews for Bluesky links.

When pasting a `bsky.app` link in a Mastodon post (for example https://bsky.app/profile/pjanuel.bsky.social/post/3lu2vr6wnn224), it correctly shows a link preview:

<img width="1194" height="472" alt="Image" src="https://github.com/user-attachments/assets/7d275098-fb6f-4cd4-9f67-0f9603907e59" />

I do not know what is happening here but it is definitely not something on the Mastodon side.

**Comment by snarfed** [comment]
Thank you @renchap! That's really useful. And also my bad here, I forgot that Mastodon generates link previews async, so when I tested a bsky.app link on https://mas.to/@snarfed2/114855093655208994 and didn't see a preview at first, I guessed that it was specific to bsky.app. Glad to hear it's not!

**Comment by snarfed** [comment]
Done in 0bf9b20482525e6746e6cf67c52d442caa4e816e!
